### ISSUE TITLE
Mapping issue: <CR> gets remapped by TOC in quickfix list

### ISSUE THREAD

**Issue by nils-gg** [nvim-neorg/neorg#514]
### Prerequisites

- [X] I have read [contribution guidelines](https://github.com/vhyrro/neorg/blob/main/docs/CONTRIBUTING.md)
- [X] I am using the latest version of the plugin
- [X] I am using either 0.6 stable or the latest compiled neovim version

### Neovim Version

v0.7.2

### Neorg setup

```
require('neorg').setup {
	-- Tell Neorg what modules to load
	load = {
		["core.defaults"] = {}, -- Load all the default modules
		--        ["core.keybinds"] = { -- Configure core.keybinds
		--          config = {
		--            default_keybinds = true, -- Generate the default keybinds
		--            --neorg_leader = "<Leader><Leader>" -- This is the default if unspecified
		--            neorg_leader = "<Leader>"
		--          }
		--        },
		["core.norg.esupports.metagen"] = {
			config = { -- Note that this table is optional and doesn't need to be provided
			-- Configuration here
                template = {
                    {
                        "title",
                        function()
                            return vim.fn.expand("%:p:t:r")
                        end,
                    },
                    { "description", "" },
                    { "authors", require("neorg.external.helpers").get_username },
                    { "categories", "" },
                    { "project", "" },
                    {
                        "created",
                        function()
                            return os.date("%Y-%m-%d")
                        end,
                    },
                    { "version", require("neorg.config").version },
                },
            }
        },
        ["core.keybinds"] = {
            config = {
                hook = function(keybinds)
                    keybinds.remap_event("norg", "n", "<leader>j", "core.presenter.next_page")
                    keybinds.remap_event("norg", "n", "<leader>k", "core.presenter.previous_page")
                end,
            }
        },
        --["core.norg.concealer"] = {}, -- Allows for use of icons
        --["core.norg.to-file"] = {}, -- Allows for use of icons
        ["core.norg.dirman"] = { -- Manage your directories with Neorg
            config = {
                workspaces = {
                    notes = "/Users/nige/notes/",
                    --gtd = "/Users/nige/gtd"
                }
            }
        },
        ["core.gtd.base"] = {
            config = {
                workspace = "notes",
            },
        },
        ["core.norg.completion"] = {
            config = {
                engine = "nvim-cmp" -- We current support nvim-compe and nvim-cmp only
            }
        },
        ["core.norg.qol.todo_items"] = {},
        ["core.norg.qol.toc"] = {
            config = { -- Note that this table is optional and doesn't need to be provided
            -- Configuration here
            default_toc_mode = "toqflist"
        }
            },
            ["core.norg.journal"] = {
                config = { -- Note that this table is optional and doesn't need to be provided
                -- Configuration here
            }
        },
        ["core.presenter"] = {
            config = { -- Note that this table is optional and doesn't need to be provided
            -- Configuration here
            --zen_mode = "truezen",
            zen_mode = "zen-mode",
        },
            },
            ["core.export"] = {
                config = { -- Note that this table is optional and doesn't need to be provided
                -- Configuration here
            },
        },
    },
}
```

### Actual behavior

When running `:Neorg toc split` in a neorg buffer, `<CR>`inside the quickfix list gets remapped from:

```
:verbose nmap <buffer> <cr>
n  <CR>         @<CR><Space>
        Last set from Lua
```

to

```
:verbose nmap <buffer> <cr>
n  <CR>        *@<Cmd>Neorg keybind norg core.norg.esupports.hop.hop-link<CR>
        Last set from Lua
```

### Expected behavior

Expected behaviour is that the toc mapping does not impact the `<CR>` mapping inside the quickfix list.

### Steps to reproduce

1. Open a neorg document containing `= TOC`
2. make a grep to populate the quickfixlist
3. verify that you can jump to files from the quick fix list via `<CR>`
4. return to the neorg document
5. `Neorg toc split`
6. Switch to quickfixlist
7. Try to jump to file via `<CR>`

### Help

No

**Comment by vhyrro** [comment]
Oof, I can reproduce. This is a severe limitation of `core.keybinds`'s integration with `core.mode`. In fact I think this is the call for me to rewrite that module entirely to use 0.7 lol. I'll let you know when I get around to fixing it

**Comment by nils-gg** [comment]
Did a little test of my own:
Tried to recreate the current mapping exactly as it is today, using `vim.keymap.set`.
But that didn't seem to work out: the new mappings do not seem to be applied...

**lua/neorg/modules/core/keybinds/module.lua line 366**

```
if type(data.command) == "string" then
    -- vim.api.nvim_buf_set_keymap(buf, mode, key, data.command, data.opts or {})
    vim.keymap.set(mode, key, data.command, 
        vim.tbl_extend(data.opts , { remap = true, buffer = buf } or {}))
elseif neorg.utils.is_minimum_version(0, 7, 0) then
    -- vim.api.nvim_buf_set_keymap(
    --     buf,
    --     mode,
    --     key,
    --     "",
    --     vim.tbl_extend("force", data.opts or {}, {
    --         callback = data.command,
    --     })
    -- )
    vim.keymap.set(
        mode,
        key,
        "",
        vim.tbl_extend("force", data.opts, { remap = true, buffer = buf } or {}, {
            callback = data.command,
        })
    )
end
```

**Comment by danymat** [comment]
From what I remember, the problem is due to the norg autocommands and the fact that how norg weirdly catches autocommands.
The aim is to only allow the keybinds to be active on a certain buffer (the toc), and deactivated on the others.

I will try to find a solution to this, and this could be a very important fix not only for TOC module, but for GTD too (the views).

**Comment by vhyrro** [comment]
Has been fixed since the TOC module rewrite as of `2.0.0`!
